### ISSUE TITLE
Which version to mine or other's units

### ISSUE THREAD

**Issue by ghost** [koying/SPMC#688]
Hi

Is there an way to find this info. I hva searched here and there, but not found any logic. Miss an litt info on the http://spmc.semperpax.com/ site. 

I have to downgrade, when there is issues as people are talking about. But then I need to know which version.

I'm running nVidia Shield TV (16GB, 1.version), and always installed from the Playstore. Now I will turn off autoupdate and install 16.4.2 (think that worked OK, have to try). 
But can't find which version I need.

And what the difference between them
ARMEABI, ARMEABI_launcher, X86, X86_launcher (X86 is normal computer I think?)

Hope for answer. 

Petter

**Comment by sammarbella** [comment]
You missed the first link in there. :)

https://github.com/koying/SPMC/releases/tag/16.5.5-spmc

First or second build will work fine.

Shield TV has a 64 bit chip, aarch64 builds are optimized for it.

Use first if you don't have root and/or don't want SPMC as Shield TV launcher (default home).

Use second if you are root AND want SPMC as Shield TV launcher (default home).

Yes x86 builds are for PCs, armV7a for 32 bit android devices.

**Comment by ghost** [comment]
Thanks for answering 😊
But I want to downgrade to 16.4.2, that worked good for me. As I can remember.
I have 16.5.4 or 16.5.5 now, and it is not good.

16.4.2 does not have aarch64, should I then use armeabi?

Petter

**Comment by sammarbella** [comment]
Use v7a (armv7a 32 bit) builds no prob.

32 bit apks works in Shield TV too, in fact +95% apps for our device are still 32 bit.

**Comment by ghost** [comment]
OK. Thanks again.

Hope info for this, will come on some page 😊

**Comment by zspeciman** [comment]
How about for firetv 1st gen, firetv 2nd gen and firetv-stick?

**Comment by sammarbella** [comment]
Amazon first gen is 32 bit, stick too.

Amazon second gen is 64 bit.

http://forum.kodi.tv/showthread.php?tid=301610

Google is our friend...

**Comment by JFG90** [comment]
Is launcher for shield?

**Comment by sammarbella** [comment]
It works fine in Shield TV having root and using a Launcher switch app like this after installing the SPMC launcher version:
https://play.google.com/store/apps/details?id=com.asad.homeswitcher&hl=en

It should work in other Android devices but i don't know for sure.

**Comment by JFG90** [comment]
I'm using spmc 16.4.2 launcher. Apk just curious as to watch launcher means

**Comment by sammarbella** [comment]
Home screen/main screen of device, in Windows is the desktop with taskbars, menus and shortcuts (icons),

In layman words an Android launcher is the desktop manager of the device, here you have a few examples:

https://play.google.com/store/apps/details?id=com.DXIDev.Top_TV_launcher&hl=en

https://play.google.com/store/apps/details?id=com.awe.dev.pro.tv

https://play.google.com/store/apps/details?id=net.i.akihiro.halauncher&hl=en


When you are root, install SPMC launcher version AND use it as launcher after you switched the launcher with a launcher switch app the Android TV main screen is SPMC itself.

**Comment by zspeciman** [comment]
has anyone compared a 32 vs 64 bit version of spmc on the shield or amazon firetv gen 2 as far as performance, response time, stability.  Is it possible to install both a 32bit and 64bit spmc on the same machine?  I wouldn't mind testing these.

**Comment by zspeciman** [comment]
On the Shield:  So I've installed the 64bit version on top of the 32bit.  The good news is all my personal setup doesn't get erased.  However, how can I tell if this is the 64bit version or not?
They both say the same build:  SPMC (16.5-5 Git:2016-12-28-972797d-dirty)
One difference is the 4 cpu vs the 2 cpu.  Is that the answer?
Attached are the 32bit before and the 64bit after, at least that what I think.
![before](https://cloud.githubusercontent.com/assets/25136898/22093326/aeed6f6a-ddb9-11e6-9e0c-840c5eaa5aa2.jpg)
![after](https://cloud.githubusercontent.com/assets/25136898/22093334/b57f7364-ddb9-11e6-8b53-322f74eb6f15.jpg)
Update:  I installed the 64bit on top of my 32bit on my amazon firetv gen2.  It had all 3cpu, before and after.  So at this point I'm not sure how to verify which version is actually installed.

**Comment by sammarbella** [comment]
Are you root?

Been root a simple way to check 32 or 64 bit for installed apps is to look inside it's installation data folder.

In SPMC case it's: /data/app/com.semperpax.spmc16(xxx)/cat

If there is a folder inside called "arm64" the app is 64 bit, if there is a folder called "arm"  the app is 32 bit.

**Comment by zspeciman** [comment]
No, I'm not root.  Does ES explorer let you navigate to these folders?  
Assuming I do have the 64bit installed, when google play auto update to the next spmc release, will I get the 32 or the 64bit version?

**Comment by sammarbella** [comment]
I don't know if Google Play updates the 64 bit SPMC  version.

You can always update manually like you did the last time.

About 2 core=32 bit and 4 cores= 64 bit it seems to work that way:

https://github.com/koying/SPMC/issues/584
